### PR TITLE
HF hub integration tweaks

### PR DESCRIPTION
**Pull request by brimoor**
- single progress bar when downloading parquet files datasets
- pass `token` to all relevant methods in `push_to_hub()`

### CODE REVIEW

File: fiftyone/utils/huggingface.py [hunk 1]
```diff
@@ -109,6 +111,10 @@ def push_to_hub(
     else:
         tags = _get_dataset_tags(dataset)
 
+    # do this now in case HF login fails before we do anything expensive
+    hf_username = hfh.whoami(token=token)["name"]
```
Comment (brimoor) [inline fiftyone/utils/huggingface.py:115]: needed to pass `token` here or else `HF_TOKEN` is strictly required

File: fiftyone/utils/huggingface.py [hunk 2]
```diff
@@ -945,53 +928,57 @@ def _add_parquet_subset_to_dataset(dataset, config, split, subset, **kwargs):
...
+        f"Downloading {num_rows} images from {config.name} ({split}, {subset})..."
     )
 
     tags = [split]
     if subset != "default" and subset != config._repo_id:
         tags.append(subset)
 
-    for start_idx in range(0, num_rows, batch_size):
-        urls_and_filepaths = []
+    with fou.ProgressBar(total=num_rows) as pb:
```
Comment (brimoor) [inline fiftyone/utils/huggingface.py:949]: single progress bar tracking the entire operation

File: fiftyone/utils/huggingface.py (hunk 1 above)
Comment (brimoor) [reply fiftyone/utils/huggingface.py:115]: @jacobmarks FYI I had an older version of `huggingface_hub` installed that didn't seem to support the `HF_TOKEN` environment variable. Can you track down the min version that supports it and update the `ensure_package()` call above to let the user know the min version that they need?

Comment (jacobmarks) [reply fiftyone/utils/huggingface.py:115]: Looks like this is where they added support in Google Colab at least: https://github.com/huggingface/huggingface_hub/pull/1895
👉 https://github.com/huggingface/huggingface_hub/releases/tag/v0.20.0
so `>=0.20.0`

**Review by jacobmarks — APPROVED** [review]
LGTM! 👍  Love the progress bar upgrade.

Will change the `huggingface_hub` version req

**Pull request merged**